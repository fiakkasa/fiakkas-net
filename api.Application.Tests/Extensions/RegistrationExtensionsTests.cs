using api.Application.Models;
using HotChocolate.Execution;

namespace api.Application.Extensions.Tests;

public class RegistrationExtensionsTests
{
    private static readonly DateTimeOffset _start = DateTimeOffset.Now;

    [Fact]
    public void AddApiApplication_Service_Registration_Should_Add_SystemInfo_With_Empty_Version_When_Version_Does_Not_Resolve()
    {
        var result =
            new ServiceCollection()
                .AddApiApplication(_start)
                .BuildServiceProvider()
                .GetService<SystemInfoItem>();

        result.Should().NotBeNull();
        result!.Version.Should().BeEmpty();
    }

    [Fact]
    public void AddApiApplication_Service_Registration_Should_Add_SystemInfo_With_Version()
    {
        var value = "1.1.1";
        var result =
            new ServiceCollection()
                .AddApiApplication(_start, new(value))
                .BuildServiceProvider()
                .GetService<SystemInfoItem>();

        result.Should().NotBeNull();
        result!.Version.Should().Be(value);
    }

    [Fact]
    public async Task AddApiApplication_GraphQL_Registration_Should_Add_GraphQL_Assets()
    {
        var result =
            await new ServiceCollection()
                .AddGraphQL()
                .AddQueryType()
                .AddApiApplication()
                .BuildSchemaAsync();

        var schema = result.Print();

        schema.Should().NotBeEmpty();

        schema.MatchSnapshot();
    }
}
