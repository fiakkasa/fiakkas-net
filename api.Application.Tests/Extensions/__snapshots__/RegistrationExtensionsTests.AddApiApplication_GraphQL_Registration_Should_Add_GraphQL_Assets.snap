schema {
  query: Query
}

type HealthReportItem {
  description: String
  duration: TimeSpan!
  status: HealthStatus!
  tags: [String!]
}

type HealthReportSummary {
  status: HealthStatus!
  totalDuration: TimeSpan!
  entries: [KeyValuePairOfStringAndHealthReportItem!]
}

type KeyValuePairOfStringAndHealthReportItem {
  key: String!
  value: HealthReportItem!
}

type Query {
  health: HealthReportSummary!
  systemStatus: SystemInfoItem!
}

type SystemInfoItem {
  version: String!
  startupTime: DateTime!
  upTime: TimeSpan!
  health: HealthReportSummary!
}

enum HealthStatus {
  UNHEALTHY
  DEGRADED
  HEALTHY
}

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime @specifiedBy(url: "https:\/\/www.graphql-scalars.com\/date-time")

"The `TimeSpan` scalar represents an ISO-8601 compliant duration type."
scalar TimeSpan
